proto/tendermint/abci/types.proto

```diff
@@ -262,6 +299,33 @@ message ResponseApplySnapshotChunk {
   }
 }
 
+message ResponsePrepareProposal {
+  bool                             modified_tx             = 1;
+  repeated TransactionRecord       txs                     = 2;
+  bytes                            app_hash                = 3;
+  repeated ExecTxResult            tx_resultsq             = 4;
+  repeated ValidatorUpdate         validator_updates       = 5 [(gogoproto.nullable) = false];
```

Comment:
Is the gogo annotation meant to imply this field cannot be empty? Or does it merely force the caller to set it empty explicitly instead of to null in the generated code?

Reply:
I don't object to including the annotation, it just stood out because it was the only one among others that could, at least in theory, have it.

I would prefer we try to avoid annotations we don't need, since they increase the semantic burden of reading the file. I also generally feel that nullability is not something that should be specified in the proto file—that's a language-specific concern, not an encoding one.

Having said that, I realize we already have a bunch of these elsewhere, and I think it affects the way gogo generates code (😢) so if we need to do it for consistency that's fine.

Reply:
@cmwaters what's your take here.
Quite honestly, I tend to agree with @creachadair that "repeated Blah" should not have the "nullable = false" tag. I'm inclined to:
* Remove it from `ResponsePrepareProposal`,  `ResponseProcessProposal`, and `ResponseFinalizeBlock`
* Leave it in `ResponseEndBlock` ... anyway we'll nuke that one soon enough
Let me know if you have concerns with this approach.